File: src/test/java/org/stt/update/VersionComparatorTest.java
package org.stt.update;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.Matchers.greaterThan;
import static org.junit.Assert.assertThat;

public class VersionComparatorTest {
    private VersionComparator sut = new VersionComparator();

    @Test
    public void moreDigitsShouldMeanHigherVersion() {
        // GIVEN

        // WHEN
        int result = sut.compare("111", "99");

        // THEN
        assertThat(result, greaterThan(0));
    }

    @Test
    public void missingVersionPartMeansLowerVersion() {
        // GIVEN

        // WHEN
        int result = sut.compare("1.1.1", "1.1");

        // THEN
        assertThat(result, greaterThan(0));
    }

    @Test
    public void sameVersionMeanEqual() {
        // GIVEN

        // WHEN
        int result = sut.compare("1.1.1", "1.1.1");

        // THEN
        assertThat(result, is(0));
    }

}

File: src/test/kotlin/org/stt/update/VersionComparatorTest.kt
package org.stt.update

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class VersionComparatorTest {
    private val sut = VersionComparator()

    @Test
    fun moreDigitsShouldMeanHigherVersion() {
        // GIVEN

        // WHEN
        val result = sut.compare("111", "99")

        // THEN
        assertThat(result).isGreaterThan(0)
    }

    @Test
    fun missingVersionPartMeansLowerVersion() {
        // GIVEN

        // WHEN
        val result = sut.compare("1.1.1", "1.1")

        // THEN
        assertThat(result).isGreaterThan(0)
    }

    @Test
    fun sameVersionMeanEqual() {
        // GIVEN

        // WHEN
        val result = sut.compare("1.1.1", "1.1.1")

        // THEN
        assertThat(result).isEqualTo(0)
    }

    @Test
    fun higherVersionShouldBeHigher() {
        // GIVEN

        // WHEN
        val result = sut.compare("2-SNAPSHOT", "1.1.1")

        // THEN
        assertThat(result).isEqualTo(1)
    }

}